src/services/user/services/AdminUsers.js

```diff
@@ -275,25 +275,28 @@ class AdminUsers {
 	}
 
 	async remove(id, params) {
-		const { _ids } = params.query;
-		const currentUser = await getCurrentUserInfo(params.account.userId);
-
+		const _ids = [];
 		if (id) {
-			const userToRemove = await getCurrentUserInfo(id);
-			if (!equalIds(currentUser.schoolId, userToRemove.schoolId)) {
-				throw new Forbidden('You cannot remove users from other schools.');
+			if (typeof id === 'object') {
```

Comment:
under what circumstances is this an object?

Reply:
It's allowed to call remove() by using an user object as parameter `id`. But even in this case, I don't know, if this condition makes much sense. Because `id` maybe is an ObjectID().

Reply:
Maybe I should return to the _original_ function

Reply:
It will look like this:
```
	async remove(id, params) {
		const { _ids } = params.query;
		const currentUser = await getCurrentUserInfo(params.account.userId);

		if (id) {
			const userToRemove = await getCurrentUserInfo(id);
			if (!equalIds(currentUser.schoolId, userToRemove.schoolId)) {
				throw new Forbidden('You cannot remove users from other schools.');
			}
			return [await this.app.service(`users/v2/admin/${this.roleName}`).remove(id, params)];
		}

		const usersIds = await Promise.all(_ids.map((userId) => getCurrentUserInfo(userId)));
		if (usersIds.some((user) => !equalIds(currentUser.schoolId, user.schoolId))) {
			throw new Forbidden('You cannot remove users from other schools.');
		}

		const removePromises = _ids.map((userId) =>
			this.app.service(`users/v2/admin/${this.roleName}`).remove(userId, params)
		);
		return Promise.all(removePromises);
	}

```